src/main/java/algorithms/curated170/medium/NumberOfCornerRectangles.java

```diff
@@ -13,21 +13,24 @@ public static void main(String[] args) {
         nocr.countCornerRectangles(grid);
     }
     
- public int countCornerRectangles(int[][] grid) {
-      int count = 0;
-      int x = 0, y = 0;
-     
-      int distance = 1;
-      while(y +distance < grid.length && x+distance < grid[y].length){
-       
-          if(grid[x][y] == 1 && grid[x+distance][y] == 1 
-                  && grid[x][y+distance] == 1 && grid[x+distance][y+distance] == 1){
-                count++;  
-             }
-        distance++;
-      
-      }
-      return count;
-      
- }
+   public int countCornerRectangles(int[][] grid) {
+        int totalCount = 0;
+        int rows = grid.length;
+        int cols = grid[0].length;
+        int[][] counts = new int[rows][cols];
```

Comment:
Why don't we set the size to int[cols][cols]? Don't we only assign values to cols here?

Reply:
Yep, you are absolutely right. Somehow, it passes all LeetCode test cases. 
It means test cases are not rigorous enough.
Thanks for your attention. We fix it.